src/indexedDb/getClearableDatabases.js

```diff
@@ -0,0 +1,15 @@
+import { dbExists } from './dbExists'
+import { dhis2DatabaseNames } from './dhis2DatabaseNames'
+
+/**
+ * @returns {Promise.<String[]>}
+ */
+export const getClearableDatabases = () =>
+    Promise.all(
+        dhis2DatabaseNames.map(name =>
+            dbExists(name).then(exists => (exists ? name : null))
+        )
+    ).then(keys => {
+        const filtered = keys.filter(key => key)
+        return filtered
+    })
```

Comment:
Just want to add that I disagree. That would make it wait for each request before processing next, instead of firing all in parallel.

Reply:
```js
export const async getClearableDatabases = () => {
    const  = await Promise.all(
        dhis2DatabaseNames.map(async name => 
            (await dbExists(name)) ? name : null
        )
    ).filter(name => !!name)
}
```

The parallel promise method can be simplified a tiny bit.  I don't think the serial method @HendrikThePendric mentioned would be problematic here, since the promises should be fast, so no strong preference between the two.